File: src/main/java/com/bajdcc/LALR1/ui/UIRemoteWindow.java
package com.bajdcc.LALR1.ui;

import com.bajdcc.LALR1.interpret.module.ModuleRemote;
import com.bajdcc.LALR1.ui.drawing.UIRemoteGraphics;

import javax.swing.*;
import java.awt.*;
import java.awt.event.WindowEvent;

/**
 * 【界面】远程用户界面
 *
 * @author bajdcc
 */
public class UIRemoteWindow extends JFrame {
	private UIRemotePanel panel;
	private static final String remoteWndTitle = "jMiniOS Remote Window";

	public UIRemotePanel getPanel() {
		return panel;
	}

	public UIRemoteWindow() {
		panel = new UIRemotePanel();
		this.setTitle(remoteWndTitle);
		this.setDefaultCloseOperation(JFrame.DO_NOTHING_ON_CLOSE);
		this.setPreferredSize(new Dimension(UIRemotePanel.w, UIRemotePanel.h));
		this.setContentPane(panel);
		this.pack();
		this.setLocationRelativeTo(null);
		this.setResizable(false);
		this.setVisible(true);
		this.setTimer();
		ModuleRemote.showMainFrame();
	}

	private void close() {
		this.dispatchEvent(new WindowEvent(this, WindowEvent.WINDOW_CLOSING));
	}

	public UIRemoteGraphics getUIGraphics() {
		return this.panel.getUIGraphics();
	}

	private void setTimer() {
		new Timer(33, e -> panel.repaint()).start();
	}
}

File: src/main/kotlin/com/bajdcc/LALR1/ui/UIRemoteWindow.kt
package com.bajdcc.LALR1.ui

import com.bajdcc.LALR1.interpret.module.ModuleRemote
import com.bajdcc.LALR1.ui.drawing.UIRemoteGraphics
import java.awt.Dimension
import java.awt.event.WindowEvent
import javax.swing.JFrame
import javax.swing.Timer

/**
 * 【界面】远程用户界面
 *
 * @author bajdcc
 */
class UIRemoteWindow : JFrame() {
    private val panel: UIRemotePanel = UIRemotePanel()

    val uiGraphics: UIRemoteGraphics
        get() = this.panel.uiGraphics

    init {
        this.title = remoteWndTitle
        this.defaultCloseOperation = JFrame.DO_NOTHING_ON_CLOSE
        this.preferredSize = Dimension(UIRemotePanel.w, UIRemotePanel.h)
        this.contentPane = panel
        this.pack()
        this.setLocationRelativeTo(null)
        this.isResizable = false
        this.isVisible = true
        this.setTimer()
        ModuleRemote.showMainFrame()
    }

    private fun close() {
        this.dispatchEvent(WindowEvent(this, WindowEvent.WINDOW_CLOSING))
    }

    private fun setTimer() {
        Timer(33) { panel.repaint() }.start()
    }

    companion object {
        private const val remoteWndTitle = "jMiniOS Remote Window"
    }
}
